pkg/api/helpers/errors.go

```diff
@@ -132,12 +133,23 @@ func ResponseAsError(response interface{}) error {
 		statusText = http.StatusText(statusCode)
 	}
 	var message string
-	f = r.FieldByName("Body")
-	if f.IsValid() && f.Type().Kind() == reflect.Slice && f.Type().Elem().Kind() == reflect.Uint8 {
-		body := f.Bytes()
-		var apiError api.Error
-		if json.Unmarshal(body, &apiError) == nil && apiError.Message != "" {
-			message = apiError.Message
+	r := reflect.Indirect(reflect.ValueOf(httpResponse))
```

Comment:
Not sure I understand this code.  Now `httpResponse` has a known type `*http.Response`.  Why do we need to reflect on it?

Reply:
Yeah. I cannot run tests on my PC due to env issues, so the only way to test is by committing and pushing to github. Please disregard any commits until it's final.